ocs_ci/ocs/scale_noobaa_lib.py

```diff
@@ -380,3 +382,65 @@ def check_all_obcs_status(namespace=None):
         else:
             obc_not_bound_list.append(status)
     return obc_bound_list, obc_not_bound_list
+
+
+def get_noobaa_pods_status():
+    """
+    Check Noobaa pod status to ensure it is in Running state.
+
+    Args: None
+
+    Returns:
+        Boolean: True, if all Noobaa pods in Running state. False, otherwise
+
+    """
+    ret_val = True
+    ocp_pod = OCP(kind=constants.POD, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+    pod_items = ocp_pod.get(selector=constants.NOOBAA_APP_LABEL).get("items")
+
+    # Check if noobaa pods are in running state
+    for nb_pod in pod_items:
+        status = ocp_pod.get_resource_status(nb_pod.get("metadata").get("name"))
+        if status == constants.STATUS_RUNNING:
+            log.info("Noobaa pod in running state...")
+        else:
+            log.error(f"Noobaa pod is in {status}, expected in Running state")
+            ret_val = False
+    return ret_val
+
+
+def check_memory_leak_in_noobaa_endpoint_log():
+    """
+    Check memory leak log in Noobaa endpoint logs.
+
+    Raises:
+        UnexpectedBehaviour: If memory leak error is existing in Noobaa endpoint logs.
+
+    """
+
+    # Collect must gather logs
+    must_gather_logs = tempfile.mkdtemp()
```

Comment:
@tiffanyn108 we dont need to take complete must-gather logs.. you can fetch endpoint logs and see if there is a memory leak

Reply:
Updated `check_memory_leak_in_noobaa_endpoint_log` to check logs for only noobaa pods.